Add note about buffer size for EventPipe (#23142)
dotnet/docs: docs/core/diagnostics/dotnet-trace.md

  Sets the size of the in-memory circular buffer, in megabytes. Default 256 MB.

  > [!NOTE]

>   > If the target process is writing events too frequently, it can overflow this buffer which results in some events to get dropped. When too many events are getting dropped, increasing the buffer size may help.

Reviewer: ```suggestion
  > If the target process writes events too frequently, it can overflow this buffer and some events might be dropped. If too many events are getting dropped, increase the buffer size.
```

Author: Thanks for the feedback @gewarren - I had to reword it a little bit because the "may help" phrasing here was chosen deliberately because it doesn't always help. I've updated the PR to reflect the suggestions minus that part and made some clarifications to that.

>   > If the target process writes events too frequently, it can overflow this buffer and some events might be dropped. If too many events are getting dropped, increase the buffer size to see if the number of dropped events reduces. If the number of dropped events does not decrease with a larger buffer size, it may be due to a slow reader blocking the target process' buffers to be flushed.

Reviewer 2: ```suggestion
  > If the target process writes events too frequently, it can overflow this buffer and some events might be dropped. If too many events are getting dropped, increase the buffer size to see if the number of dropped events reduces. If the number of dropped events does not decrease with a larger buffer size, it may be due to a slow reader preventing the target process' buffers from being flushed.
```
